plugin/src/create-schema-customization.ts

```diff
@@ -1,4 +1,55 @@
-import { CreateSchemaCustomizationArgs } from "gatsby";
+import {
+  CreateSchemaCustomizationArgs,
+  NodePluginSchema,
+  GatsbyGraphQLObjectType,
+} from "gatsby";
+
+function addFields(
```

Comment:
This is pure bikeshedding so feel free to ignore this 😆 but this feels like premature abstraction since it's spreading an object without doing anything else to the fields.

```js
    addFields(productDef, {
      collections: {
        type: `[${name("ShopifyCollection")}]`,
        extensions: {
          link: {
            from: "id",
            by: "productIds",
          },
        },
      },
    });
```

vs

```js
    productDef.config.fields = {
      ...(productDef.config.fields || {}),
      collections: {
        type: `[${name("ShopifyCollection")}]`,
        extensions: {
          link: {
            from: "id",
            by: "productIds",
          },
        },
      },
    }
```

It's about the same amount of code but without the abstraction you can see immediately what the code is doing without checking the `addFields` fn.

Reply:
You might be right. It's currently used twice and is mainly so that I have an API where I don't have to remember that `config.fields` is optional or worry about how to get around that. It would be great if the core APIs themselves made this easier.

Reply:
You might be right. It's currently used twice and is mainly so that I have an API where I don't have to remember that `config.fields` is optional or worry about how to get around that. It would be great if the core APIs themselves made this easier.

Reply:
That's a good point, in the end it's super minor and not essential at all. Nice work on this plugin btw :)